onnxruntime/contrib_ops/cpu/cdist.cc

```diff
@@ -10,7 +15,94 @@ namespace contrib {
                                 KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<data_type>()), \
                                 CDist<data_type>);
 DEFINE_KERNEL(float);
-DEFINE_KERNEL(double);
+//DEFINE_KERNEL(double);
+
+template <typename T>
+static void cdist_euclidian_onshot(const Tensor& a, const Tensor& b, Tensor& c, concurrency::ThreadPool* threadpool) {
+  // input shapes have already been validated
+  const auto& shape_a = a.Shape().GetDims();  // {m, k}
+  const auto& shape_b = b.Shape().GetDims();  // {n, k}
+  int64_t m = shape_a[0];
+  int64_t n = shape_b[0];
+  int64_t k = shape_a[1];
+
+  // https://github.com/droyed/eucl_dist/wiki/Main-Article
+  // dist(Xi,Yj) = sum_k(Xik**2) + sum_k(Yjk**2) - 2*sum_k(Xik*Yjk)
+
+  const auto* a_data = a.Data<T>();
+  const auto* b_data = b.Data<T>();
+  auto* c_data = c.MutableData<T>();
+
+  // ReduceSumSquare for A
+  std::vector<T> a_ss;
+  a_ss.reserve(m);
+  const auto* cur_a = a_data;
+  for (int64_t i = 0; i < m; ++i) {
```

Comment:
i is not used, maybe Worth to remove it and use cur_a as the loop variable.

Reply:
Is comparison between cur_a and a float* or double* for the end of the data cheaper than using int64_t  for the counter?

Reply:
I guess it should be the same with a int64_t pointer, even faster if data is aligned I assume but i was thinking you need to update one less variable.

Reply:
Instead of looking at the cost of "i", consider sizing a_ss to the correct number of elements up front `(std::vector<T> a_ss(m)` and then replacing push_back with the simpler a_ss[i] = squaredNorm(). As it is, each push_back is calling out to something way more expensive than i vs not i. Avoiding the call lets the compiler keep more in registers that otherwise are destroyed by the call. You shift some cost to the zeroing of the buffer in the initial std::vector resize, something to measure.

Reply:
Depending on the model the difference can be extremely minimal to slightly noticeable, however in this scenario the GEMM cost far outweights anything happening in the steps involving the vector.  

Using resize() instead of push_back() is better in my testing. Using a pointer to the data instead of operator[] is slightly better again. And if you're using a pointer instead of operator[] it doesn't matter if you use 'i' for the loop or compare to the end pointer for either the input or output buffer (same number of registers and instructions). 

The zeroing cost of the resize() could be avoided if the TempSpaceAllocator is used instead of leveraging std::vector for the memory management, but that may come with other costs around what happens when IAllocator::Free is called. I'll keep it simple and not make that change for now given the cost of the resize() call isn't significant.